### ISSUE TITLE
[Bug Report][3.0.0-alpha.9] It's not possible to use an icon with a colon in its name/path/data

### ISSUE THREAD

**Issue by glen-84** [vuetifyjs/vuetify#13975]
### Environment
**Vuetify Version:** 3.0.0-alpha.9
**Vue Version:** 3.1.5
**Browsers:** Chrome 92.0.4515.107
**OS:** Windows 10

### Steps to reproduce
See reproduction.

While working on the reproduction for #13974, I noticed that if you have a custom icon set that accepts image paths (for example), then it'll fail if the path contains a colon (f.e. in the scheme of a URL).

```html
<v-icon icon="img:https://example.com/image.svg">
```

With this code, the set would be correctly extracted as "img", but the icon would be set to just "https".

A simple solution would be to just extract from the first colon to the end of the string, however that will fail for default icon sets:

```html
<v-icon icon="https://example.com/image.svg">
```

Named icon set ("https") or default icon set?

Dealing with formatted strings is often tricky, so I'm wondering if it wouldn't make more sense to use a prop for the set, like:

```html
<v-icon set="img" icon="https://example.com/image.svg">
```

### Expected Behavior
Ability to use colons in icon name/path/data.

### Actual Behavior
Not possible.

### Reproduction Link
<a href="https://codepen.io/glen-84/pen/KKmVOmB" target="_blank">https://codepen.io/glen-84/pen/KKmVOmB</a> (from #13974, where I was forced to use a root-relative URL)

**Comment by glen-84** [comment]
@nekosaur What do you think? I'm not able to create an icon set that accepts URLs without this change. 😞

**Comment by nekosaur** [comment]
Yeah I actually had `set` and `icon` as separate props in an earlier iteration of VIcon, but @johnleider didn't like it :P

Finding a separator that won't collide with potential values might be tricky..

**Comment by glen-84** [comment]
I've just realized that the reason for this might be to allow icons to be passed to other props like:

```html
<avatar icon="mdi:foo" />
```

If the `set` is a prop, then you'd have to have something like:

```html
<avatar icon-set="mdi" icon="foo" />
```

... which might not be ideal.


> One way to keep `:` could be to explicitly check for registered sets with `startsWith`

I hadn't thought of this solution. I feel like it'll work in 99.9% of cases, which is probably good enough. 🙂

Let me know if I can help in some way. I'd love to be able to create a URL-based icon set soon.

**Comment by nekosaur** [comment]
You're welcome to do a PR. Otherwise I'll try to get to it soon.
